### ISSUE TITLE
Unable to merge features in QField

### ISSUE THREAD

**Issue by Paul-Losse** [opengisch/QField#4967]
### Describe the issue
When attempting to merge adjacent polygons in QField, the merge button is greyed out.

### Reproduction steps
digitise two adjacent polygons. Select one, long press on the feature to select it. Select another polygon. Select 'Merge selected feature(s). This option is greyed out

### Expected behavior
'Merge selected features' button is not greyed out. Pressing the button results in merging polygons

### Observed behavior
'Merge selected features' button is greyed out and not selectable. Unable to merge features

### Screenshots and GIFs
[If applicable, add screenshots and GIFs to help explain your problem.]

[Please also attach additional files if a specific project/dataset is useful to investigate the problem.]

### Desktop (please complete the following information)
 - OS: Windows 10
 - QGIS Version [e.g. 3.22.0]
 - QFieldSync Version [e.g.4.0.0]

### Mobile (please complete the following information)
 - Device: Galaxy Tab Active 3
 - OS: Android 13
 - QField version: 3.1.9- Borneo (f2829b v3.1.9 

### Additional information
* Problem started happening recently, didn't happen in an older version of QField: Yes
* Problem can be reliably reproduced, doesn't happen randomly: Yes
* Problem happens with all files and projects, not only some files or projects: Yes

[If the problem happened with QFieldCloud, please add your username and project name.]
![20240119_110625](https://github.com/opengisch/QField/assets/77330795/c63e5ea5-3741-4d73-a1f2-e848e5403114)

**Comment by nirvn** [comment]
@Paul-Losse , merging of polygons require the following conditions (https://github.com/opengisch/QField/blob/windows/src/core/multifeaturelistmodelbase.cpp#L330-L337):
- editable vector layer
- multi-type geometry (i.e. multipolygon in this case)
- data provider's ability to delete features and modify geometries

Is it possible your polygon layer is *not* multipolygon? If so, we can transform this as a feature request to support non multi-type geometry merging.

**Comment by m-kuhn** [comment]
Should be documented here https://docs.qfield.org/how-to/digitize/#merging-features @SeqLaz can you open a doc ticket?

**Comment by Paul-Losse** [comment]
You are correct. The polygon layer is not multipolygon. It would be great if merging could support non-multi-type geometry merging (I assumed it already did). Many thanks

**Comment by nirvn** [comment]
Implemented in the latest version of QField 🥳
